### PR TITLE
M5415 App manager

### PR DESCRIPTION
**Pull request by mark-de-haan**
Documentation is WIP, opening PR for sonar information and travis build

#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits
- [ ] Added Feature/Fix to release notes
- [ ] Integration tests run correctly

### CODE REVIEW

File: molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/AppDeployService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	void loadJavascriptResources(String uri, String fileName, HttpServletResponse response) throws IOException;
+
+	/**
+	 * Load css files from the file store
+	 *
+	 * @param uri      An uri belonging to an app
+	 * @param fileName File name of the requested css file
+	 * @param response a {@link HttpServletResponse} used to attach any found css files
+	 */
+	void loadCSSResources(String uri, String fileName, HttpServletResponse response) throws IOException;
```
Comment (fdlk) [inline molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/AppDeployService.java]: To keep this service independent of http servlet spec I'd let this method yield an InputStream and then let the controller copy it to the response. Also makes for easier testing.

File: molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppDeployServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+	public AppDeployServiceImpl(DataService dataService)
+	{
+		this.dataService = requireNonNull(dataService);
+	}
+
+	@Override
+	public String configureTemplateResourceReferencing(String template, String baseUrl)
+	{
+		template = template.replaceAll("src=js/", "src=" + baseUrl + "js/");
```
Comment (fdlk) [inline molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppDeployServiceImpl.java]: but what if they DO add "" in their html?

File: molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppDeployServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+
+import static java.net.URLConnection.guessContentTypeFromName;
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class AppDeployServiceImpl implements AppDeployService
+{
+	private static final String JS_FOLDER = "js";
```
Comment (fdlk) [inline molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppDeployServiceImpl.java]: Determine mime type based on file extension (just like ordinary spring resources are served from classpath). 
If you pick app serve URI in a way that doesn't clash with the meta methods you can serve all folders.

File: molgenis-app-manager/src/main/frontend/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# molgenis-app-manager-ui
+
+> UI code for the app-manager plugin
+
+## Build Setup
+
+``` bash
+# install dependencies
+npm install
```
Comment (ConnorStroomberg) [inline molgenis-app-manager/src/main/frontend/README.md]: we recently updated this to yarn , it would be good if we update this

**Pull request closed without merging**